Lib/pdb.py

```diff
@@ -1332,6 +1332,11 @@ def do_list(self, arg):
         if last is None:
             last = first + 10
         filename = self.curframe.f_code.co_filename
+        # gh-93696: stdlib frozen modules provide a useful __file__
+        # this workaround can be removed with the closure of gh-89815
+        if filename.startswith("<frozen"):
+            if "__file__" in self.curframe.f_globals:
+                filename = self.curframe.f_globals["__file__"]
```

Comment:
What if `__file__` is not a string? It looks like `get_file_breaks` will crash in that case.

Reply:
Hmm, that's a good point. I would think it's a small edge, but it's easy to be more cautious about it.

Reply:
Yes, I think it's important for a debugger to be resilient in the face of unusual program state.